packages/calcite-components/src/components/link/link.scss

```diff
@@ -90,11 +86,11 @@ calcite-icon {
   // icon positioning and styles
   &.icon-start {
     margin-inline-end: theme("margin.2");
-    color: var(--calcite-link-icon-start-color);
+    color: var(--calcite-link-icon-start-color, currentColor);
   }
   &.icon-end {
     margin-inline-start: theme("margin.2");
-    color: var(--calcite-link-icon-end-color);
+    color: var(--calcite-link-icon-end-color, currentColor);
```

Comment:
Per Franco's comment, let's make the icon color one prop `--calcite-link-icon-color`